### PR TITLE
[FEATURE] Reference Environment - BigQuery

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What is in this Reference Environment PR?
- Docker file, which requires 2 ENV variables
   - `GOOGLE_APPLICATION_CREDENTIALS` which refers to the credential configuration JSON file.
   - `BIGQUERY_CONNECTION_STRING` which allows GX to connect to the BigQuery project and dataset. 
- Jupyter Notebook
- Changes to the cli so that `great_expectations example bigquery` will spin up the appropriate environment
- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: examples/reference_environments/bigquery/compose.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    build:
+      context: .
+      dockerfile: jupyter.Dockerfile
+    environment:
+      GOOGLE_APPLICATION_CREDENTIALS: /tmp/creds.json
+      BIGQUERY_CONNECTION_STRING: ${BIGQUERY_CONNECTION_STRING}
+    ports:
+      - "8888:8888"
+    volumes:
+      - ${GOOGLE_APPLICATION_CREDENTIALS}:/tmp/creds.json
```
Comment (Shinnnyshinshin) [inline examples/reference_environments/bigquery/compose.yaml:14]: Mapping the local `.json` file to the Docker container without copying it. 

* Pattern from [the following StackOverflow](https://stackoverflow.com/questions/75645717/where-to-securely-store-google-application-credentials-in-django-project)

File: examples/reference_environments/bigquery/jupyter.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM jupyter/minimal-notebook:python-3.10
+
+WORKDIR /gx
+
+COPY ./bigquery_example.ipynb ./
+
+# sqlalchemy-bigquery connector is currently not compatible with sqlachemy 2.0.
```
Comment (anthonyburdi) [inline examples/reference_environments/bigquery/jupyter.Dockerfile:7]: ❤️ Thank you for the helpful comment!

File: great_expectations/cli/example.py [hunk 3]
```diff
@@ -65,6 +65,11 @@ def example_snowflake(
     rebuild: bool,
 ) -> None:
     """Start a snowflake database example."""
+    unset_env_vars = _check_snowflake_env_vars()
+    if unset_env_vars:
+        cli_message(
+            f"<red>Please check your config, currently we only support connecting via env vars. You are missing the following vars: {', '.join(unset_env_vars)}</red>"
+        )
```
Comment (anthonyburdi) [inline great_expectations/cli/example.py:72]: 🎉

**Review by anthonyburdi — APPROVED** [review]
Awesome! Retrieving the credentials json file from the local machine seems like the best approach. Thank you for implementing this!

**Pull request merged**